### ISSUE TITLE
Interface Orientation

### ISSUE THREAD

**Issue by CooperRS** [TeehanLax/UIViewController-Transitions-Example#1]
Custom transition is broken in landscape orientation

**Comment by ashfurrow** [comment]
Yeah, I'm waiting to hear back on Apple about this, since it's a larger problem than just our project. Interface rotation is broken for all non-portrait presentations. Probably a result of the iPad beta not being ready for WWDC. Take a look at [Stack Overflow](http://stackoverflow.com/search?tab=relevance&pagesize=50&q=custom%20transition%20rotation) to see just how many people are having problems with this. 

I'll update the project when I find a suitable solution.

**Comment by CooperRS** [comment]
Already had the problem with my own implementation. I first though it was me being too stupid, but then I will have to wait for Apple.

Thanks for your answer!

**Comment by CooperRS** [comment]
I think using autolayout may be a possible solution to this problem. Could not test it though...

**Comment by skirsche** [comment]
Here is where I found how to make  it work in landscape orientation:
http://www.brightec.co.uk/blog/ios-7-custom-view-controller-transitions-and-rotation-making-it-all-work

The linked article contains this response from Apple:

> "For custom presentation transitions we setup an intermediate view between the window and the windows rootViewController's view. This view is the containerView that you perform your animation within. Due to an implementation detail of auto-rotation on iOS, when the interface rotates we apply an affine transform to the windows rootViewController's view and modify its bounds accordingly. Because the containerView inherits its dimensions from the window instead of the root view controller's view, it is always in the portrait orientation."
> 
> "If your presentation animation depends upon the orientation of the presenting view controller, you will need to detect the presenting view controller's orientation and modify your animation appropriately. The system will apply the correct transform to the incoming view controller but you're animator need to configure the frame of the incoming view controller."
